tests/firedrake/adjoint/test_hessian.py

```diff
@@ -179,7 +174,7 @@ def test_nonlinear():
     Jhat = ReducedFunctional(J, Control(f))
 
     h = Function(V)
-    h.vector()[:] = 10*rng.random(V.dim())
+    h.dat.data_wo[...] = 10*rng.random(h.dat.data_wo.shape)
```

Comment:
We should make this tests deterministic
```suggestion
    rg = RandomGenerator(PCG64(seed=1234))
    h = rg.unform(V, -10, 10)
```